misc/python/materialize/cloudtest/k8s/environmentd.py

```diff
@@ -0,0 +1,168 @@
+    V1PodTemplateSpec,
+    V1ResourceRequirements,
+    V1SecretVolumeSource,
+    V1Service,
+    V1ServicePort,
+    V1ServiceSpec,
+    V1StatefulSet,
+    V1StatefulSetSpec,
+    V1Volume,
+    V1VolumeMount,
+)
+from pg8000 import Cursor
+
+from materialize.cloudtest.k8s import K8sService, K8sStatefulSet
+
+
+class EnvironmentdService(K8sService):
+    def __init__(self) -> None:
+        service_port = V1ServicePort(name="sql", port=6875)
+        self.service = V1Service(
+            api_version="v1",
+            kind="Service",
+            metadata=V1ObjectMeta(name="environmentd", labels={"app": "environmentd"}),
+            spec=V1ServiceSpec(
+                type="NodePort", ports=[service_port], selector={"app": "environmentd"}
+            ),
+        )
+
+
+class EnvironmentdStatefulSet(K8sStatefulSet):
+    def __init__(self) -> None:
+
+        metadata = V1ObjectMeta(name="environmentd", labels={"app": "environmentd"})
+        label_selector = V1LabelSelector(match_labels={"app": "environmentd"})
+
+        value_from = V1EnvVarSource(
+            field_ref=V1ObjectFieldSelector(field_path="metadata.name")
+        )
+
+        env = [
+            V1EnvVar(name="MZ_POD_NAME", value_from=value_from),
+            V1EnvVar(name="AWS_REGION", value="minio"),
+            V1EnvVar(name="AWS_ACCESS_KEY_ID", value="minio"),
+            V1EnvVar(name="AWS_SECRET_ACCESS_KEY", value="minio123"),
+        ]
+
+        ports = [V1ContainerPort(container_port=5432, name="sql")]
+
+        volume_mounts = [
+            V1VolumeMount(name="data", mount_path="/data"),
+            V1VolumeMount(name="secrets-mount", mount_path="/secrets"),
+        ]
+
+        s3_endpoint = urllib.parse.quote("http://minio-service.default:9000")
+
+        container = V1Container(
+            name="environmentd",
+            image=self.image("environmentd"),
+            args=[
+                "--storaged-image=" + self.image("storaged"),
+                "--computed-image=" + self.image("computed"),
+                f"--persist-blob-url=s3://minio:minio123@test/test?endpoint={s3_endpoint}&region=minio",
+                "--orchestrator=kubernetes",
+                "--orchestrator-kubernetes-image-pull-policy=never",
+                "--persist-consensus-url=postgres://postgres@postgres.default?options=--search_path=consensus",
+                "--adapter-stash-url=postgres://postgres@postgres.default?options=--search_path=catalog",
+                "--storage-stash-url=postgres://postgres@postgres.default?options=--search_path=storage",
+                "--unsafe-mode",
+            ],
+            env=env,
+            ports=ports,
+            volume_mounts=volume_mounts,
+        )
+
+        volumes = [
+            V1Volume(
+                name="secrets-mount",
+                secret=V1SecretVolumeSource(
+                    default_mode=400, secret_name="user-managed-secrets"
+                ),
+            )
+        ]
```

Comment:
This can get ditched; I forgot to remove it from cluster-dev. (Similar comment below.)